### DISCUSSION TITLE
Programmatically create an organization codespace secret

### DISCUSSION THREAD

**Discussion by lime-green** [community/community#8570] [Codespaces] [answered]
Hi there, I am using a base image hosted on an AWS ECR repository therefore I need to authenticate via setting `ECR_CONTAINER_REGISTRY_PASSWORD`. The token generated from `aws ecr get-login-password ` lasts for 12 hours therefore I am wanting to create an automated solution to update this secret at the organization level. I've searched the docs (https://docs.github.com/en/rest/reference/codespaces) and it doesn't seem like creating or updating codespace secrets at the org-level is supported. Is there a better way to approach this problem? If not, is the API going to support updating an org codespace secret soon?

Cheers!

**Comment by asciimike** [comment, marked as answer]
@lime-green, apologies for the delay here, but in https://github.com/github/feedback/discussions/5873 we came up with a solution to ECR that's a little more elegant: https://docs.github.com/en/codespaces/codespaces-reference/allowing-your-codespace-to-access-a-private-image-registry#accessing-aws-elastic-container-registry
